src/actions/sources/select.js

```diff
@@ -90,15 +91,21 @@ export function selectSource(sourceId: string, tabIndex: string = "") {
  * @memberof actions/sources
  * @static
  */
-export function selectLocation(location: Location, tabIndex: string = "") {
+export function selectLocation(
+  location: Location,
+  tabIndex: string = "",
+  preserveOriginalSource: boolean = false
```

Comment:
re: `preserveOriginalSource`
Lets see what else we can do here...

maybe selectLocation needs to differentiate between the first time we are going to the source where we 
a) might want to create the pretty source
b) need to setSymbols, setOutOfScopeLocations

and selecting a source in general. 

Lastly, could we move the decision for which source to another function because *that* is complicated. Here are two ideas:

1. source tree on click checks to see if the source has a pretty file and if so calls selectLocation with that source
2. source tree calls a function like `selectBestLocation` where that function checks to see if there is a pretty file and if so calls that... i dunno it's hard as we discussed :)

Reply:
I somewhat agree that `selectLocation` is doing too much; the problem is that `selectLocation` gets called from *many* different places:

1.  SourceTree
2.  Tab (individual tab click)
3.  Quick open / search
..... and so on

The only two spots we *know* which tab (original or pretty) should be selected is clicking the individual tab from the tab bar and clicking the footer link from the pretty tab -- they should both be original.  In all other cases, we will want to go with the prefs.autoPrettyPrint choice, right?  Just thinking out loud.